shift_detector/checks/SimpleCheck.py

```diff
@@ -0,0 +1,161 @@
+        return {"categorical_comparison": categorical_comparison, 'numerical_comparison': numerical_comparison}
+
+    @staticmethod
+    def compare_numerical_columns(df1, df2):
+        numerical_comparison = dict()
+        empty_metrics_dict = {'mean': {}, 'median': {}, 'min': {}, 'max': {}, 'quartile_1': {}, 'quartile_3': {},
+                              'uniqueness': {}, 'distinctness': {}, 'completeness': {}, 'std': {}, 'available_in': {}}
+
+        for df_name, df in [('df1', df1), ('df2', df2)]:
+            for column in df.columns:
+                if df_name == 'df1':
+                    numerical_comparison[column] = deepcopy(empty_metrics_dict)
+                elif not numerical_comparison.get(column):
+                    numerical_comparison[column] = deepcopy(empty_metrics_dict)
+
+                numerical_comparison[column]['available_in'][df_name] = True
+
+                # TODO Later Vielleicht: verschnellerbar, in dem man alle Quantile gleichzeitig berechnet,
+                #  also quantile([0, 0.25,  ... ]) oder Methoden selbst berechnet
+                numerical_comparison[column]['min'][df_name] = df[column].min()
+                numerical_comparison[column]['max'][df_name] = df[column].max()
+                numerical_comparison[column]['quartile_1'][df_name] = df[column].quantile(.25)
+                numerical_comparison[column]['quartile_3'][df_name] = df[column].quantile(.75)
+
+                numerical_comparison[column]['median'][df_name] = df[column].median()
+                numerical_comparison[column]['mean'][df_name] = df[column].mean()
+
+                column_droppedna = df[column].dropna()
+                numerical_comparison[column]['std'][df_name] = column_droppedna.std()
+
+                numerical_comparison[column]['distinctness'][df_name] = column_droppedna.nunique() / \
+                                                                        len(column_droppedna)
+
+                numerical_comparison[column]['completeness'][df_name] = len(column_droppedna) / len(df1[column])
+
+                numerical_comparison[column]['uniqueness'][df_name] = len(df.groupby(column)
+                                                                    .filter(lambda x: len(x) == 1)) / \
+                                                                    len(column_droppedna)
+        return numerical_comparison
+
+    @staticmethod
+    def compare_categorical_columns(df1, df2):
+        category_comparison = {}
+        for column in df1.columns:
+            category_comparison[column] = {}
+            attribute_ratios = df1[column].value_counts(normalize=True).to_dict()
+            for key, value in attribute_ratios.items():
+                category_comparison[column][key] = {}
+                category_comparison[column][key]['df1'] = value
+
+        for column in df2.columns:
+            if not category_comparison.get(column):
```

Comment:
This should not happen

Reply:
I left it, makes the code more robust, even if it _should_ not happen